scrapers_next/ca/committees_test.py

```diff
@@ -0,0 +1,335 @@
+                mem_name, mem_role = mem.split(",")
+                # mem_name = mem_name.strip()
+                # mem_role = mem_role.strip()
+                if "(" in mem_role:
+                    mem_role = mem_role.lstrip("(").rstrip(")")
+                if "of the" in mem_role:
+                    mem_role = mem_role.split("of the")[0]
+                if mem_name == "Kevin Kiley":
+                    mem_role = "Vice Chair"
+                # print(mem_name, mem_role)
+            elif re.search(r"\s\((V|C)", mem):
+                mem_name, mem_role = mem.split("(")
+                # mem_name = mem_name.strip()
+                mem_role = mem_role.rstrip(")")
+                # print(mem_name, mem_role)
+            elif re.search(r"\n", mem):
+                mem_name, mem_role = mem.split("\n")
+                # mem_name = mem_name.strip()
+                mem_role = mem_role.split("of the")[0]
+            elif mem_num == 0:
+                mem_name = mem.strip()
+                mem_role = "Chair"
+                # print(member, "chair not listed")
+            else:
+                mem_name = mem.strip()
+                mem_role = "Vice Chair"
+                # print(member, "vice chair not listed")
+            mem_num += 1
+
+            # print(mem_name.strip(), mem_role.strip())
+            com.add_member(
+                mem_name.strip(), role=mem_role.strip() if mem_role else "member"
+            )
+
+        # print("TYPE FOUR")
+        return com
+
+
+class SenateCommitteeList(HtmlListPage):
+    source = URL("http://senate.ca.gov/committees")
+
+    selector = XPath("//h2/../following-sibling::div//a")
+
+    def process_item(self, item):
+        comm_name = XPath("text()").match_one(item)
+        if comm_name in ["Teleconference How-To Information", "Legislative Process"]:
+            self.skip()
+
+        comm_url = XPath("@href").match_one(item)
+
+        if comm_name.startswith("Joint"):
+            # self.skip()
+            com = ScrapeCommittee(
+                name=comm_name, classification="committee", parent="legislature"
+            )
+        elif comm_name.startswith("Subcommittee"):
+            com = ScrapeCommittee(
+                name=comm_name, classification="subcommittee", parent=""
+            )
+        else:
+            com = ScrapeCommittee(
+                name=comm_name, classification="committee", parent="upper"
+            )
+        com.add_source(self.source.url)
+        com.add_source(comm_url)
+        com.add_link(comm_url, note="homepage")
+        return ChooseType(com, source=URL(comm_url))
+
+
+class AssemblyCommitteeList(HtmlListPage):
+    source = URL("https://www.assembly.ca.gov/committees")
+
+    selector = CSS("div .block.block-views ul li", num_items=98)
+
+    def process_item(self, item):
+        comm_name = CSS("a").match_one(item).text_content()
+        comm_url = CSS("a").match_one(item).get("href")
+
+        if (
```

Comment:
I wonder if there is a better way to do this, to make it more readable

if not I'd at least add a comment sort of documenting the page structure around this that made the many parent calls required